### PR TITLE
Verify_pass_on_XRT_2024.2

### PR DESCRIPTION
**Pull request by karthdmg-xilinx**

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_xgq.c [hunk 1]
```diff
@@ -288,6 +288,11 @@ int xocl_xgq_attach(struct xocl_xgq *xgq_handle, void *client, struct semaphore
 
 	spin_lock_irqsave(&xgq->xx_lock, flags);
 
+	if (!xgq) {
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_xgq.c:291]: Which situation can cause the xgq is NULL?

File: src/runtime_src/core/pcie/driver/linux/include/mailbox_proto.h [hunk 2]
```diff
@@ -85,6 +85,8 @@ enum xcl_mailbox_request {
 	XCL_MAILBOX_REQ_SDR_DATA =		16,
 	XCL_MAILBOX_REQ_LOAD_XCLBIN_SLOT_KADDR = 17,
 	XCL_MAILBOX_REQ_LOAD_SLOT_XCLBIN =	18,
+	XCL_MAILBOX_REQ_PROTOCOL_VERSION =	19,
+	XCL_MAILBOX_REQ_GET_XCLBIN_UUID =		20,
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/include/mailbox_proto.h:89]: indent issue?

File: src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c [hunk 3]
```diff
@@ -1245,6 +1245,16 @@ void xclmgmt_mailbox_srv(void *arg, void *data, size_t len,
 		vfree(resp);
 		break;
 	}
+#if 0
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-core.c:1248]: Why if 0 here, we should check the protocol version.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c [hunk 4]
```diff
@@ -297,12 +297,12 @@ static bool verify_timestamp(struct platform_device *pdev, u64 timestamp)
 	rom = platform_get_drvdata(pdev);
 	BUG_ON(!rom);
 
-	xocl_dbg(&pdev->dev, "Shell timestamp: 0x%llx",
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/feature_rom.c:300]: We possibly need to make our own feature_com.c and call it as a different subdev name. We don't want to modify the existing feature_rom.c.

I think we can create a subdev called feature_rom_versal.c, we can copy the code we use from the feature_rom.c

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xdma.c [hunk 5]
```diff
@@ -564,6 +564,8 @@ static int xdma_remove(struct platform_device *pdev)
 			if (irq_entry->enabled) {
 				xocl_err(&pdev->dev,
 					"ERROR: Interrupt %d is still on", i);
+					xocl_info(&pdev->dev, "Forcing user IRQ unregistration");
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xdma.c:567]: we don't use xdma? we use qdma I guess.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c [hunk 6]
```diff
@@ -2591,9 +2591,11 @@ static int __icap_download_bitstream_user(struct platform_device *pdev,
 		goto done;
 
 	xocl_subdev_destroy_by_slot(xdev, slot_id);
-
+/* workaround 1: skip peer download */
+#if 1
 	err = __icap_peer_xclbin_download(icap, xclbin, slot_id);
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c:2596]: maybe, when mailbox_version is 1, skip peer download.
instead use check_uuid to see if peer is already download the xclbin.

I think that was the original design.

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h [hunk 7]
```diff
@@ -656,6 +656,15 @@ struct xocl_dev_core {
 
 	/* XOCL Should cache some of the information shared in IOCTL */
 	struct xocl_axlf_obj_cache *axlf_obj[MAX_SLOT_SUPPORT];
+	struct timer_list	mgmt_status_timer;
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h:659]: I don't think we need timer in both common.h and xocl_drv.h.
if possible, we should only add timer in the xocl_dev.

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h [hunk 8]
```diff
@@ -656,6 +656,15 @@ struct xocl_dev_core {
...
+	struct work_struct	mgmt_status_poll;
+	u32			mbx_protocol_version;
+	bool			is_mbx_version_valid;
+	struct platform_device	*dma_platdev;
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h:663]: we might think to name those platdev as vmgmt_mbx_platdev thus people knows that this is just for vmgmt to use.

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_xgq.c (hunk 1 above)
Comment (karthdmg-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/xocl_xgq.c:291]: not sure, I added this when pulling changes from Nishad's commit. removing this now t align with latest XRT code base as it is not related to vmgmt change

File: src/runtime_src/core/pcie/driver/linux/include/mailbox_proto.h (hunk 2 above)
Comment (karthdmg-xilinx) [reply src/runtime_src/core/pcie/driver/linux/include/mailbox_proto.h:89]: addressing this in the new commit

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c (hunk 6 above)
Comment (karthdmg-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c:2596]: do we have to change the existing code logic here?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xdma.c (hunk 5 above)
Comment (karthdmg-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/xdma.c:567]: agree, reverting the change

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h (hunk 7 above)
Comment (karthdmg-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h:659]: agree, removed in common.h

File: src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h (hunk 8 above)
Comment (karthdmg-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/xocl_drv.h]: agree, changing in the next commit

**Comment by karthdmg-xilinx** [comment]
Raising another commit merging past 3 commits

**Pull request closed without merging**